pkg/occlient/occlient_test.go

```diff
@@ -4343,3 +4343,173 @@ func TestWaitAndGetDC(t *testing.T) {
+						"app.kubernetes.io/component-name": "ruby",
+						"app.kubernetes.io/component-type": "ruby",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				envVars: []corev1.EnvVar{
+					{
+						Name:  "key",
+						Value: "value",
+					},
+					{
+						Name:  "key1",
+						Value: "value1",
+					},
+				},
+			},
+			wantErr: true,
+			actions: 1,
+		},
+		{
+			name: "Case 3 - Generate and create the BuildConfig but fail with unable to parse image name",
+			args: args{
+				builderImage: "::",
+				namespace:    "testing",
+				gitURL:       "https://github.com/openshift/ruby",
+				commonObjectMeta: metav1.ObjectMeta{
+					Name: "ruby",
+					Labels: map[string]string{
+						"app":                              "apptmp",
+						"app.kubernetes.io/component-name": "ruby",
+						"app.kubernetes.io/component-type": "ruby",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				envVars: []corev1.EnvVar{
+					{
+						Name:  "key",
+						Value: "value",
+					},
+					{
+						Name:  "key1",
+						Value: "value1",
+					},
+				},
+			},
+			wantErr: true,
+			actions: 1,
+		},
+		{
+			name: "Case 4 - Generate and create the BuildConfig and pass in no envVars",
+			args: args{
+				builderImage: "ruby:latest",
+				namespace:    "testing",
+				gitURL:       "https://github.com/openshift/ruby",
+				commonObjectMeta: metav1.ObjectMeta{
+					Name: "ruby",
+					Labels: map[string]string{
+						"app":                              "apptmp",
+						"app.kubernetes.io/component-name": "ruby",
+						"app.kubernetes.io/component-type": "ruby",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				envVars: []corev1.EnvVar{},
+			},
+			wantErr: false,
+			actions: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, fakeClientSet := FakeNew()
+
+			fakeClientSet.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStream(tt.args.commonObjectMeta.Name, tt.args.commonObjectMeta.Namespace, []string{"latest"}), nil
+			})
+
+			// Run function CreateBuildConfig
+			_, err := fakeClient.CreateBuildConfig(tt.args.commonObjectMeta, tt.args.builderImage, tt.args.gitURL, tt.args.envVars)
+
+			// Error checking CreateBuildConfig
+			if !tt.wantErr == (err != nil) {
```

Comment:
I guess this condition is satisfied by the last two checks. This will print the two error messages indicating the same thing
```
=== RUN   TestCreateBuildConfig/Case_1_-_Generate_and_create_the_BuildConfig
    --- FAIL: TestCreateBuildConfig/Case_1_-_Generate_and_create_the_BuildConfig (0.00s)
        occlient_test.go:4499:  client.CreateBuildConfig() unexpected error sdka, wantErr false
        occlient_test.go:4510: test failed, expected: no error, got error: sdka
```